### PR TITLE
Ko/add auth for icarus httpsless

### PR DESCRIPTION
**Pull request by ibihim**

### CODE REVIEW

File: example/tabs/login.js [hunk 1]
```diff
@@ -1,9 +1,23 @@
...
     var loginResultElement = document.getElementById('loginResults');
+}
+
+var gcLoginBtn = $("#gcLoginBtn");
+
+function disableGcLoginBtn() {
+    gcLoginBtn.prop("disabled", true)
+}
+
+GC.AUTH.config({
```
[the remark this answers is not in the document]
Comment (ibihim) [reply example/tabs/login.js]: main page, ok :+1:

File: example/tabs/login.js [hunk 2]
```diff
@@ -1,9 +1,23 @@
 /* eslint-disable no-undef */
-if (typeof loginResultElement === 'undefined')
+if (typeof loginResultElement === 'undefined') {
     var loginResultElement = document.getElementById('loginResults');
+}
+
+var gcLoginBtn = $("#gcLoginBtn");
+
+function disableGcLoginBtn() {
```
[the remark this answers is not in the document]
Comment (ibihim) [reply example/tabs/login.js]: ok ok

File: src/authCloud/index.js [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        sdk = window.GC.SDK,
+        globalWindow = window,
+    }) {
+        const postKeys = ({ privateKey, publicKey }) =>
+            request.post(`${clientURL}/gckeys`)
+                .send({ private_key: privateKey, public_key: publicKey });
+
+        const getToken = () =>
+            request.get(`${clientURL}/gctoken`).then(({ body }) => {
+                const resObj = body || {};
```
Comment (Glumli) [inline src/authCloud/index.js]: I also get an undefined body back here

File: example/server.js [hunk 4]
```diff
@@ -1,10 +1,20 @@
-let express = require('express');
-
-let app = express();
-let host = '0.0.0.0';
-let port = 8888;
+const express = require('express');
+const cors = require('cors');
```
Comment (Harsimran1) [inline example/server.js]: We are not really using `cors` here. But we might need it, the the domain on which icarus is deployed is different from the one in web app

File: src/authCloud/index.js [hunk 5]
```diff
@@ -0,0 +1,93 @@
...
+                throw err;
+            });
+
+        const capPromise = sdk.createCAP();
+        const setupHC = getToken().then(({ privateKey, token }) => {
+            if (isNotEmpty(token)) {
+                return sdk.setup(
+                    clientId,
+                    privateKey,
+                    onceElse(
```
Comment (Harsimran1) [inline src/authCloud/index.js]: would be nice to comment on what this method is doing. Can be a little confusing :P

File: src/authCloud/index.js [hunk 6]
```diff
@@ -0,0 +1,93 @@
...
+        clientId,
+        clientURL,
+        sdk = window.GC.SDK,
+        globalWindow = window,
+    }) {
+        const postKeys = ({ privateKey, publicKey }) =>
+            request.post(`${clientURL}/gckeys`)
+                .send({ private_key: privateKey, public_key: publicKey });
+
+        const getToken = () =>
```
Comment (Harsimran1) [inline src/authCloud/index.js]: Would be nice to rename this method to something else as its also returning `privateKey`

File: test/authCloud/indexTest.js [hunk 7]
```diff
@@ -0,0 +1,141 @@
+/* eslint-disable no-unused-vars */
+/* eslint-disable no-unused-expressions */
+/* eslint-disable max-nested-callbacks */
+import 'babel-polyfill';
+import chai, { expect } from 'chai';
+import sinon from 'sinon';
+import proxy from 'proxyquireify';
```
Comment (Harsimran1) [inline test/authCloud/indexTest.js]: not using `proxyquireify`

**Comment by ibihim** [comment]
rewrite of commit history

**Comment by ibihim** [comment]
@Harsimran1 @Glumli PTAL

**Pull request merged**